atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -1316,15 +1317,22 @@ private ConcurrentNavigableMap<Cell, byte[]> getLocalWrites(TableReference table
 
     /**
      * This includes deleted writes as zero length byte arrays, be sure to strip them out.
+     *
+     * For the selectedColumns parameter, empty set means all columns. This is unfortunate, but follows the semantics of
+     * {@link RangeRequest}.
      */
-    private SortedMap<Cell, byte[]> getLocalWritesForRange(TableReference tableRef, byte[] startRow, byte[] endRow) {
+    private SortedMap<Cell, byte[]> getLocalWritesForRange(
+            TableReference tableRef, byte[] startRow, byte[] endRow, SortedSet<byte[]> selectedColumns) {
         SortedMap<Cell, byte[]> writes = getLocalWrites(tableRef);
         if (startRow.length != 0) {
             writes = writes.tailMap(Cells.createSmallestCellForRow(startRow));
         }
         if (endRow.length != 0) {
             writes = writes.headMap(Cells.createSmallestCellForRow(endRow));
         }
+        if (!selectedColumns.isEmpty()) {
+            writes = Maps.filterKeys(writes, cell -> selectedColumns.contains(cell.getColumnName()));
```

Comment:
Things I checked for: `Maps::filterKeys` returns a new map, rather than modifying. `SortedSet::contains` handles `byte[]` correctly, which it looks like it does (well, ImmutableSortedSet does). 

Also validated that `selectedColumns` should return all when empty.

Reply:
Sweet! Yeah that's the case